frontend/src/lib/components/Dropdown/dropdown.tsx

```diff
@@ -14,26 +14,35 @@ import { IconButton } from "../IconButton";
 import { Input } from "../Input";
 import { Virtualization } from "../Virtualization";
 
+export type DropdownOptionGroup<TValue = string> = {
+    label: string;
+    // ? Should groups also be allowed an adornment?
+    adornment?: never;
+    options: DropdownOption<TValue>[];
+};
+
 export type DropdownOption<TValue = string> = {
     value: TValue;
     label: string;
     adornment?: React.ReactNode;
     hoverText?: string;
     disabled?: boolean;
-    group?: string;
 };
 
+type OptionOrGroup<TValue> = DropdownOption<TValue> | DropdownOptionGroup<TValue>;
```

Comment:
Should we stick to the typing `DropdownOption<TValue> | DropdownOptionGroup<TValue>`? As `OptionOrGroup` is not exported, the users will need to write `DropdownOption<TValue> | DropdownOptionGroup<TValue>` if building options anyways?